### ISSUE TITLE
[Question] Visibility of AuthenticatedHttpClientHandler and AuthenticatedParameterizedHttpClientHandler

### ISSUE THREAD

**Issue by Elfocrash** [reactiveui/refit#705]
So let me give you some background. 

---

### Background

I am using Refit in the `WebApplicationFactory` of .NET Core to run in-memory integration tests against my API.

By default if I do `appFactory.CreateClient()` I will get a client back to access the `TestServer`. After that I do `RestService.For<MyApi>(httpClient)` and I get the api client interface back. However if my endpoint has an auth header and I provide that for my refit interface by doing `RestService.For<MyApi>(httpClient, settingsWithTheAuthProviderImplemented)` when my `AuthorizationHeaderValueGetter` in the will be ignored because the delegating handler in never added to handle this behavior.

The workaround is to use the `appFactory.CreateDefaultClient(delegatingHandlers)` alternative of `WebApplicationFactory` which allows me to add custom delegating handlers. That way I can add the `AuthenticatedParameterizedHttpClientHandler` and auth header resolution works.

---

### Question

My question is, is there any specific reason that only `AuthenticatedParameterizedHttpClientHandler` is `public` but `AuthenticatedHttpClientHandler` is not? Their behavior looks almost identical, but the first one also has the request as a paramater. Is that intended? Is the `AuthenticatedHttpClientHandler` obsolete? I feels like both of them could be `public.`

**Comment by benjaminhowarth1** [comment]
I added the `AuthenticatedParameterizedHttpClientHandler` because @onovotny said that if we changed the method signature on `AuthenticatedHttpClientHandler`, then we'd have a breaking API change.
The `AuthenticatedParameterizedHttpClientHandler` passes the HttpRequestMessage in as a parameter, in case your auth mechanism requires information about the request being made (see Twitter's API as an example). The `AuthenticatedHttpClientHandler` has no information about the HttpRequestMessage, so frankly, in my view, at some point it should be retired out of the API together, probably in a v5 release.

Technically speaking, neither classes need to be public, as they're automatically added using the `RefitSettings.AuthorizationHeaderValueGetter` / `RefitSettings.AuthorizationHeaderValueWithParamGetter` properties. If you use these, the classes are [constructed automatically by RestService.cs](https://github.com/reactiveui/refit/blob/1272833e88329493b711e79d3bd399c308b96797/Refit/RestService.cs#L88)
